skbio/io/format/_blast.py

```diff
@@ -32,10 +33,28 @@ def _parse_blast_data(fh, columns, error, error_message, comment=None,
     read_csv = functools.partial(pd.read_csv, na_values='N/A', sep='\t',
                                  header=None, keep_default_na=False,
                                  comment=comment, skiprows=skiprows)
-    lineone = read_csv(fh, nrows=1)
 
-    if len(lineone.columns) != len(columns):
-        raise error(error_message % (len(columns), len(lineone.columns)))
+    # HACK for https://github.com/pandas-dev/pandas/issues/14418
+    # this avoids closing the `fh`, whose lifetime isn't the responsibility
+    # of this parser
+    with noop_close(fh) as fh:
 
-    fh.seek(0)
-    return read_csv(fh, names=columns, dtype=_possible_columns)
+        lineone = read_csv(fh, nrows=1)
+
+        if len(lineone.columns) != len(columns):
+            raise error(error_message % (len(columns), len(lineone.columns)))
+
+        fh.seek(0)
+
+        return read_csv(fh, names=columns, dtype=_possible_columns)
+
+
+# HACK for https://github.com/pandas-dev/pandas/issues/14418
+@contextlib.contextmanager
+def noop_close(fh):
```

Comment:
Will defer as to whether this should be paired with a unit test. It's straight forward enough and exercised broadly by the test framework implicitly, so am okay without one.